client/engine/engine.go

```diff
@@ -548,6 +607,48 @@ func (e *Engine) attemptProgress(objective protocols.Objective) (outgoing Engine
 	return
 }
 
+// handleNotifications takes an objective and constructs notifications for any related channels for that objective.
+func (e *Engine) handleNotifications(o protocols.Objective) (EngineEvent, error) {
+	outgoing := EngineEvent{}
+
+	for _, rel := range o.Related() {
+		switch c := rel.(type) {
+		case *channel.Channel:
+			// If we're in direct funding/defunding then we're dealing with a leadger channel
+			_, isDF := o.(*directfund.Objective)
+			_, isDDF := o.(*directdefund.Objective)
+			if isDDF || isDF {
+				outgoing.LedgerChannelUpdates = append(outgoing.LedgerChannelUpdates, query.ConstructLedgerFromChannel(c))
+			} else { // otherwise we must have a payment channel
+
+				paid, remaining, err := e.getVoucherBalance(c.Id)
+				if err != nil {
+					return outgoing, err
+				}
+
+				info, err := query.ConstructPaymentInfo(c, o, paid, remaining)
+				if err != nil {
+					return outgoing, err
+				}
+
+				outgoing.PaymentChannelUpdates = append(outgoing.PaymentChannelUpdates, info)
+
+			}
+		case *consensus_channel.ConsensusChannel:
+
+			// TODO: Some of the related consensus channels are zeroed out (possibly deleted?)
+			// For now we just ignore them
```

Comment:
One idea would be to push the `IsZero` check inside the `objective.Related()` function. That would align with our principle of pushing complexity into objectives (which can know about things like zero channels sometimes existing etc) and out of the engine (which should be as dumb and simple as possible).

Reply:
Yup, I'll create an issue. 

> One idea would be to push the `IsZero` check inside the `objective.Related()` function. That would align with our principle of pushing complexity into objectives (which can know about things like zero channels sometimes existing etc) and out of the engine (which should be as dumb and simple as possible).

I agree, this should be handled internally in the objective. I suspect this is a bug (maybe with marshalling?) where we write out a zeroed channel instead of ignoring it.